### PR TITLE
ISSUE-410 Add utility parser function for webhook event

### PR DESCRIPTION
**Pull request by okue**
Utility parser function should be out of `line-bot-servlet` module.

This is a preparation for #410.

### CODE REVIEW

File: line-bot-parser/build.gradle [hunk 1]
```diff
@@ -0,0 +1,6 @@
+dependencies {
```
Comment (imasahiro) [inline line-bot-parser/build.gradle]: Add copyright please

File: line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class WebhookParser {
+    private final ObjectMapper objectMapper = ModelObjectMapper.createNewObjectMapper();
+    private final LineSignatureValidator lineSignatureValidator;
+
+    /**
+     * Create new instance.
```
Comment (imasahiro) [inline line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java]: `Creates a new instance.`

File: line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * @param lineSignatureValidator LINE messaging API's signature validator
+     */
+    public WebhookParser(@NonNull LineSignatureValidator lineSignatureValidator) {
+        this.lineSignatureValidator = lineSignatureValidator;
+    }
+
+    /**
+     * Parse request.
```
Comment (imasahiro) [inline line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java]: `Parses a request.`

File: line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+     *
+     * @throws WebhookParseException There's an error around signature.
+     */
+    public CallbackRequest handle(String signature, byte[] payload) throws IOException, WebhookParseException {
+        // validate signature
+        if (signature == null || signature.isEmpty()) {
+            throw new WebhookParseException("Missing 'X-Line-Signature' header");
+        }
+
+        log.debug("got: {}", new String(payload, StandardCharsets.UTF_8));
```
Comment (imasahiro) [inline line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java]: Could you wrap with `if(log.isDebugEnabled()){ ... }` not to convert to string if logger is disabled?
```
if (!log.isDebugEnabled()) {
}
```

File: settings.gradle [hunk 5]
```diff
@@ -24,3 +24,5 @@ include 'line-bot-cli'
 include 'sample-spring-boot-echo'
 include 'sample-spring-boot-echo-kotlin'
 include 'sample-spring-boot-kitchensink'
+include 'line-bot-parser'
+
```
Comment (imasahiro) [inline settings.gradle]: Remove this line

Comment (imasahiro) [reply settings.gradle]: Also, could you move to L6 because this parser is not sample code?

File: line-bot-parser/build.gradle [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+    implementation project(':line-bot-api-client')
```
Comment (kazuki-ma) [inline line-bot-parser/build.gradle]: Looks like parser & client relationship is a little difficult.

Can we make parser independent from client? (only depends for model.)

File: line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java (hunk 4 above)
Comment (okue) [reply line-bot-parser/src/main/java/com/linecorp/bot/parser/WebhookParser.java]: Thanks for a nice advice.

**Pull request merged**